### ISSUE TITLE
Tree is not visible: "ERR Bad progress location: workbench.view.explorer"

### ISSUE THREAD

**Issue by borekb** [letmaik/vscode-git-tree-compare#51]
I'm seeing a strange issue when sometimes, the tree is not visible:

<img width="356" alt="Screenshot 2019-07-23 at 10 29 01" src="https://user-images.githubusercontent.com/101152/61695965-b77f7900-ad34-11e9-911e-58f0f5cd017e.png">

Trying to invoke Refresh yields this error in VSCode notifications:

> Updating the git tree failed: The "path" argument must be of type string. Received type undefined

From the Output panel:

```
Looking for git in: /usr/local/bin/git
Using git 2.22.0 from /usr/local/bin/git
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
Updating the git tree failed: The "path" argument must be of type string. Received type undefined
> git rev-parse --git-dir
> git rev-parse --git-common-dir
Using repository: /Users/borekb/dev/shoptet/shoptet-frontend
Updating refs
> git symbolic-ref --short HEAD
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
Using stored base ref: origin/master
> git merge-base origin/master 211-component-for-complex-l18n
HEAD ref updated: undefined -> 211-component-for-complex-l18n
HEAD ref commit updated: undefined -> 323c3e0ecf6fab85188511de266d586eb681c860
Base ref updated: undefined -> origin/master
Merge base updated: undefined -> c3b8116dd3b068652be5d252ef85423e02836011
> git update-index --refresh -q
> git diff-index --no-renames c3b8116dd3b068652be5d252ef85423e02836011 --
> git ls-files --others --exclude-standard
13 diff entries
```

Sometimes, it works (for example, after I've reloaded VSCode for a couple of times, or when I tried on a different repo), and it also works reliably in VSCode Insiders it seems, so I'm not sure what's going on.

**Comment by letmaik** [comment]
The next time this happens, could you have a look at the Developer Tools console output? (see Help) With some luck there should be a backtrace for the error.

**Comment by borekb** [comment]
It just happened again, I opened the Developer Tools console and there is one line mentioning git-tree-compare (the last one). Not sure whether the before it is related or not but here it goes:

<details><summary>📜  Console output</summary>

```
console.ts:137 [Extension Host] (node:43408) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ console.ts:137
log.ts:173   ERR Bad progress location: workbench.view.explorer: Error: Bad progress location: workbench.view.explorer
    at Proxy.withProgress (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5111:20)
    at G.activate (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4802:140)
    at G.setVisibility (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4796:799)
    at z.updateTreeVisibility (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4792:324)
    at _register.onDidChangeBodyVisibility (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4791:730)
    at d.fire (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:81:67)
    at z.setVisible (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4785:52)
    at panels.filter.map.t (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4811:822)
    at Array.map (<anonymous>)
    at e.setVisible (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4811:813)
    at e.showComposite (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5100:754)
    at e.doOpenComposite (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5098:591)
    at e.openComposite (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5098:123)
    at e.doOpenViewlet (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5108:446)
    at e.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5107:827)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at e.openViewlet (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5107:744)
    at e.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5167:824)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at e.run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5167:373)
    at runAction (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:100:369)
    at r.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:100:234)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:100:97)
    at e.onClick (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:864:519)
    at Immediate.options.options.isMenu.i.setImmediate (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:864:75)
    at runCallback (timers.js:694:18)
    at tryOnImmediate (timers.js:665:5)
    at processImmediate (timers.js:647:5)
log.ts:173   ERR Bad progress location: workbench.view.explorer: Error: Bad progress location: workbench.view.explorer
    at Proxy.withProgress (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5111:20)
    at q.t.instantiationService.createInstance.e [as withProgress] (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4797:979)
    at q.getChildren (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4803:583)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:981:31)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at a.createCancelablePromise (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:984)
    at Object.l [as createCancelablePromise] (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:102:428)
    at X.doGetChildren (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:956)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:586)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at X.doRefreshNode (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:373)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:231)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at X.doRefreshSubTree (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:162)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:980:9)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at X.refreshNode (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:979:810)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:979:570)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at X.refreshAndRenderNode (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:979:523)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:976:913)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at X.updateChildren (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:976:682)
    at X.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:976:476)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at X.setInput (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:976:292)
    at G.createTree (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4799:268)
    at G.activate (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4802:106)
    at G.setVisibility (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4796:799)
    at z.updateTreeVisibility (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4792:324)
    at _register.onDidChangeBodyVisibility (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4791:730)
    at d.fire (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:81:67)
    at z.setVisible (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4785:52)
    at panels.filter.map.t (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4811:822)
    at Array.map (<anonymous>)
    at e.setVisible (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:4811:813)
    at e.showComposite (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5100:754)
    at e.doOpenComposite (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5098:591)
    at e.openComposite (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5098:123)
    at e.doOpenViewlet (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5108:446)
    at e.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5107:827)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at e.openViewlet (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5107:744)
    at e.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5167:824)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at e.run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:5167:373)
    at runAction (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:100:369)
    at r.<anonymous> (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:100:234)
    at Generator.next (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:460
    at new Promise (<anonymous>)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:35:237)
    at run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:100:97)
    at e.onClick (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:864:519)
    at Immediate.options.options.isMenu.i.setImmediate (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:864:75)
    at runCallback (timers.js:694:18)
    at tryOnImmediate (timers.js:665:5)
    at processImmediate (timers.js:647:5)
console.ts:137 [Extension Host] /Users/borekb/.vscode/extensions/letmaik.git-tree-compare-1.7.0/resources/icons
```

</details>

**Comment by borekb** [comment]
Reloading window (**Developer: Reload Window**) didn't help. What did help was **Open Recent...**, switch to another workspace, there, the Git Tree was fine, so I switched again to my main workspace and all was suddenly alright.

When everything is alright, I no longer see the long log in the devtools, specifically, the

> ERR Bad progress location: workbench.view.explorer: Error: Bad progress location: workbench.view.explorer

message is gone. My log output now ends with the deprecation warning about `Buffer()` (the first line of the log posted above).

**Comment by letmaik** [comment]
I have a feeling that this is a VS Code bug, possibly on macOS only, at least I've never seen it on Windows. You may have better luck reporting this on the VS Code issue tracker.

**Comment by borekb** [comment]
Thanks, posted as https://github.com/microsoft/vscode/issues/79371.

**Comment by borekb** [comment]
Workaround when this happens: quit VSCode entirely and start it again. **Developer: Reload Window** is _not_ enough.

**Comment by supoved** [comment]
It happened on windows as well

**Comment by VazMike** [comment]
Tree is not showing for me as well. Here's the Dev Tools log:

![image](https://user-images.githubusercontent.com/14861435/72786697-436c5200-3c3f-11ea-9ce7-951e5bdcd616.png)

[-1579594405343.log](https://github.com/letmaik/vscode-git-tree-compare/files/4089870/-1579594405343.log)

**Comment by lsdev14** [comment]
> 
> 
> Tree is not showing for me as well. Here's the Dev Tools log:
> 
> ![image](https://user-images.githubusercontent.com/14861435/72786697-436c5200-3c3f-11ea-9ce7-951e5bdcd616.png)
> 
> [-1579594405343.log](https://github.com/letmaik/vscode-git-tree-compare/files/4089870/-1579594405343.log)

I'm getting the same problem. I'm not sure, but I think it started happening after the last update from git on windows, 2.25.0

**Comment by letmaik** [comment]
@VazMike @lsdev14 This is a different problem although with the same effect. I'll move that into a separate issue.

**Comment by letmaik** [comment]
I released a new version for 1.43 (currently Insiders) which should fix this issue. See the end of https://github.com/microsoft/vscode/issues/79371 for details.

**Comment by HDCharles** [comment]
I ran into this issue when running on a remote server. None of the above advice was able to solve it, though I did eventually fix the problem and wanted to document it here. The problem was that the mounted workspace folder was something like /home/<username>/local/<repo> but 'local' is a symlink to /data/users/<username> which must have been causing an issue. So I tried instead mounting /data/users/<username>/<repo> which made the git tree compare extension start working again.
